component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java

```diff
@@ -188,4 +205,273 @@ public static String convertWindowsFileTimeToUnixTime(String windowsFileTime) {
         long millisSinceEpoch = (fileTime / HUNDREDS_OF_NANOSECONDS) - WINDOWS_EPOCH_DIFF;
         return String.valueOf(millisSinceEpoch);
     }
+
+    /**
+     * Get password expiry rules.
+     *
+     * @param tenantDomain Tenant domain.
+     * @return List of password expiry rules.
+     * @throws AuthenticationFailedException If an error occurred while getting the password expiry rules.
+     */
+    @SuppressFBWarnings("CRLF_INJECTION_LOGS")
+    public static List<PasswordExpiryRule> getPasswordExpiryRules(String tenantDomain)
+            throws AuthenticationFailedException {
+
+        List<PasswordExpiryRule> passwordExpiryRules = new ArrayList<>();
+        try {
+            IdentityGovernanceService governanceService =
+                    PasswordPolicyDataHolder.getInstance().getIdentityGovernanceService();
+            ConnectorConfig connectorConfig =
+                    governanceService.getConnectorWithConfigs(tenantDomain, CONNECTOR_CONFIG_NAME);
+            if (connectorConfig == null) {
+                return passwordExpiryRules;
+            }
+            Property[] properties = connectorConfig.getProperties();
+            if (properties == null) {
+                return passwordExpiryRules;
+            }
+
+            for (Property property : properties) {
+                if (StringUtils.startsWith(property.getName(), PASSWORD_EXPIRY_RULES_PREFIX) &&
+                        StringUtils.isNotEmpty(property.getValue())) {
+                    try {
+                        PasswordExpiryRule passwordExpiryRule = new PasswordExpiryRule(property.getValue());
+                        passwordExpiryRules.add(passwordExpiryRule);
+                    } catch (Exception e) {
+                        // Log and skip the rule if an error occurred while parsing the rule, without failing the
+                        // authentication flow.
+                        if (log.isDebugEnabled()) {
```

Comment:
Do we really need these two logs here, can't we improve the error log or may be converting error log to a warn

Reply:
resolved with https://github.com/wso2-extensions/identity-outbound-auth-passwordPolicy/pull/61/commits/a7e572e7ffc256388e0c5aa3856bd328af3a4c93